File: src/Page/SelectedStudentRequestPage.java
package Page;

import SystemMessage.FunctionType;
import SystemMessage.SystemMessage;

public class SelectedStudentRequestPage extends Page{
	
	public SelectedStudentRequestPage(String content) {
		super(content);
		setType(PageType.SELECTED_STUDENT_REQUEST_PAGE);
		setName("Selected Student Request Page");
	}

	@Override
	public SystemMessage runPage() {
		showContent();		
		String input;
		
		while (true) {
			input = takeInput();

			if (input.toLowerCase().equals("y")) {
				return new SystemMessage(FunctionType.APPROVE_REQUEST, PageType.EVALUATE_REQUESTS_PAGE, null);
			}
			
			if (input.toLowerCase().equals("r")) {
				return new SystemMessage(FunctionType.DISAPPREOVE_REQUEST, PageType.EVALUATE_REQUESTS_PAGE, null);
			}
			
			if (input.toLowerCase().equals("q")) {
				return new SystemMessage(FunctionType.CHANGE_PAGE, PageType.EVALUATE_REQUESTS_PAGE, null);
			}
			else {
				System.out.println("Wrong Input");
				continue;
			}
		}
	}
}

File: iter3/python_development/Page/SelectedStudentRequestPage.py
from iter3.python_development.Page.Page import Page
from iter3.python_development.Page.PageType import PageType
from iter3.python_development.System.FunctionType import FunctionType
from iter3.python_development.System.SystemMessage import SystemMessage


class SelectedStudentRequestPage(Page):
     
    def __init__(self, content):
        super().__init__(content)
        self.setType("SELECTED_STUDENT_REQUEST_PAGE")
        self.setName("Selected Student Request Page")


    def runPage(self):
        self.showContent()
        inputFromPage = None
        while(True):
            inputFromPage = self.takeInput()
            if(inputFromPage.lower() == "a"):
                print("Type your message or press enter directly:")
                message = self.takeInput()
                return SystemMessage("APPROVE_REQUEST", "EVALUATE_REQUESTS_PAGE", message)
            elif (inputFromPage.lower() == "r"):
                print("Type your message or press enter directly:")
                message = self.takeInput()
                return SystemMessage("DISAPPREOVE_REQUEST", "EVALUATE_REQUESTS_PAGE", message)
            elif (inputFromPage.lower() == "q"):
                return SystemMessage("CHANGE_PAGE", "EVALUATE_REQUESTS_PAGE", None)
            else:
                print("Wrong Input!")
